trusted-content: restructure doi section (#19745)
docker/docs: content/trusted-content/official-images/contributing.md

---
title: Contributing to Docker Official Images
description: |
  This article describes how Docker Official Images are created,
  and how you can contribute or leave feedback.
keywords: docker official images, doi, contributing, upstream, open source
---

Docker, Inc. sponsors a dedicated team that's responsible for reviewing and
publishing all content in Docker Official Images. This team works in
collaboration with upstream software maintainers, security experts, and the
broader Docker community.

While it's preferable to have upstream software authors maintaining their
Docker Official Images, this isn't a strict requirement. Creating
and maintaining images for Docker Official Images is a collaborative process. It takes

> place openly on GitHub where participation is encouraged. Anyone can provide

Reviewer: ```suggestion
place [openly on GitHub](https://github.com/docker-library/official-images) where participation is encouraged. Anyone can provide
```

feedback, contribute code, suggest process changes, or even propose a new
Official Image.

> **Note**
>

> > Docker Official Images are an intellectual property of Docker.

Reviewer: ```suggestion
> Docker Official Images are the intellectual property of Docker.
```

What are we trying to communicate here? Is this necessary to state?

Author: Not sure about the background for this. I will check if we really need this callout, I'd prefer to remove it to. Just need to check where it came from, so leaving this for a followup.

Reviewer 2: Yeah this one stood out to me as well, I'm sure it's been some request at some point to add it

Reviewer 3: This note was added based on a request from our Legal counsel

Reviewer 4: https://github.com/docker-library/official-images/issues/14346 🙈👀

Reviewer: @usha-mandya can you provide more context please so we can respond to given https://github.com/docker-library/official-images/issues/14346 ?

Reviewer 3: @whalelines Sure. I've shared the original email request that came from Legal and copied Joel for guidance on this.

## Creating a Docker Official Image

From a high level, an Official Image starts out as a proposal in the form
of a set of GitHub pull requests. The following GitHub repositories detail the proposal requirements:

> - [Docker Official Images repository on GitHub](https://github.com/docker-library/official-images)

Reviewer: ```suggestion
- [Docker Official Images repository on GitHub](https://github.com/docker-library/official-images#readme)
```

Author: Updated for both links

subjective concerns boil down to the basic question: "is this image generally
useful?" For example, the [Python](https://hub.docker.com/_/python/)
Docker Official Image is "generally useful" to the larger Python developer
community, whereas an obscure text adventure game written in Python last week is
not.

Once a new proposal is accepted, the author is responsible for keeping
their images up-to-date and responding to user feedback. The Official
Repositories team becomes responsible for publishing the images and
documentation on Docker Hub. Updates to the Docker Official Image follow the same pull request process, though with less review. The Docker Official Images team ultimately acts as a gatekeeper for all changes, which helps mitigate the risk of quality and security issues from being introduced.

## Submitting feedback for Docker Official Images

All Docker Official Images contain a **User Feedback** section in their
documentation which covers the details for that specific repository. In most
cases, the GitHub repository which contains the Dockerfiles for an Official
Repository also has an active issue tracker. General feedback and support

> questions should be directed to `#docker-library` on [Libera.Chat IRC](https://libera.chat).

Reviewer 4: Is it worth a dedicated channel, or should we just direct folks to `#general` (since they'll often post there anyhow, and their questions are not usually specific to DOI after all)?

As an example, with `##docker` on Libera.Chat being spiritually similar to `#general` on Slack, we don't get much _user_ chatter in `#docker-library` (mostly just maintainers looking for help, advice, or pre-review).

> questions should be directed to `#docker-library` on [Libera.Chat IRC](https://libera.chat).

Reviewer 4: I guess what I'm saying is that it might make more sense to point users to `#general` and make this note specific to maintainers?

---
title: Contributing to Docker Official Images
description: |
  This article describes how Docker Official Images are created,
  and how you can contribute or leave feedback.
keywords: docker official images, doi, contributing, upstream, open source
---

Docker, Inc. sponsors a dedicated team that's responsible for reviewing and
publishing all content in Docker Official Images. This team works in
collaboration with upstream software maintainers, security experts, and the
broader Docker community.

While it's preferable to have upstream software authors maintaining their
Docker Official Images, this isn't a strict requirement. Creating
and maintaining images for Docker Official Images is a collaborative process.
It takes place [openly on GitHub](https://github.com/docker-library/official-images)
where participation is encouraged. Anyone can provide feedback, contribute
code, suggest process changes, or even propose a new Official Image.

> **Note**
>

> > Docker Official Images are the intellectual property of Docker.

Reviewer 4: So it doesn't get lost, we do have a community request to dig more into this: https://github.com/docker-library/official-images/issues/14346

---
title: Contributing to Docker Official Images
description: |
  This article describes how Docker Official Images are created,
  and how you can contribute or leave feedback.
keywords: docker official images, doi, contributing, upstream, open source
---

Docker, Inc. sponsors a dedicated team that's responsible for reviewing and
publishing all content in Docker Official Images. This team works in
collaboration with upstream software maintainers, security experts, and the
broader Docker community.

While it's preferable to have upstream software authors maintaining their
Docker Official Images, this isn't a strict requirement. Creating
and maintaining images for Docker Official Images is a collaborative process.
It takes place [openly on GitHub](https://github.com/docker-library/official-images)
where participation is encouraged. Anyone can provide feedback, contribute
code, suggest process changes, or even propose a new Official Image.

> **Note**
>

> > Docker Official Images are the intellectual property of Docker.

Reviewer: ```suggestion
> Use of Docker Official Images is subject to [Docker's Terms of Service](https://www.docker.com/legal/docker-terms-service/).
```

This update has been approved.

Reviewer 3: ✅

Author: Thanks!

community, whereas an obscure text adventure game written in Python last week is
not.

Once a new proposal is accepted, the author is responsible for keeping their
images and documentation up-to-date and responding to user feedback. Docker is
responsible for building and publishing the images on Docker Hub. Updates to
Docker Official Images follow the same pull request process as for new images,
although review process for updates is more streamlined. The Docker Official
Images team ultimately acts as a gatekeeper for all changes, which helps
ensures consistency, quality, and security.

## Submitting feedback for Docker Official Images

All Docker Official Images contain a **User Feedback** section in their
documentation which covers the details for that specific repository. In most
cases, the GitHub repository which contains the Dockerfiles for an Official
Repository also has an active issue tracker.

General feedback and support questions about Docker Official Images
should be directed to the `#general` channel in the [Docker Community Slack](https://dockr.ly/comm-slack).

If you're a maintainer or contributor to Docker Official Images and you're

> looking for help or advice, use the `#docker-library` on [Libera.Chat IRC](https://libera.chat).

Reviewer 4: ```suggestion
looking for help or advice, use the `#docker-library` channel on [Libera.Chat IRC](https://libera.chat).
```
